osin/v1/types.go

```diff
@@ -404,8 +404,10 @@ type TokenConfig struct {
 	// consecutive uses of the token. Tokens become invalid if they are not
 	// used within this temporal window. The user will need to acquire a new
 	// token to regain access once a token times out.
-	// If this value is not set, then tokens are valid until their expiry.
-	// Takes valid time duration string such as "5m", "1.5h" or "2h45m".
+	// Takes valid time duration string such as "5m", "-1s", "1.5h" or "2h45m".
+	// Duration >= 0s: Tokens time out if there is no activity for this duration. Allowed minimum value is `300s` (5 minutes).
+	// Duration = nil: Tokens time out is disabled (default).
+	// Duration < 0s: Tokens time out is enabled but tokens never timeout unless configured per client (e.g. `-1s`).
```

Comment:
I don't think negative values are valid here

Reply:
As per https://github.com/openshift/origin/blob/28e888d76e5d104be284e0aa545e830d9e7bf3ef/vendor/k8s.io/kubernetes/openshift-kube-apiserver/authentication/oauth/timeoutvalidator.go#L110-L133 they actually behave just the same as 0